### ISSUE TITLE
Error importing large dataset

### ISSUE THREAD

**Issue by pbaylis** [sjewo/readstata13#22]
I'm importing a large dataset from Stata, and getting the following error:
"Error: When attempting to read \<strls\>: Something went wrong!"

I've included a MWE using Stata code and R code to reproduce the error below. If I use saveold in Stata, I am able to import the data using read.dta from foreign without issue (my current workaround). The machine I'm using has 192G of RAM and Stata 13, with version 0.8 of readstata13 installed.

--- STATA CODE TO GENERATE DATASET ---
clear
set obs 1000000000
gen x = 0
save "test", replace

--- R CODE ---
library(readstata13)
df <- read.dta13("test.dta")

**Comment by JanMarvin** [comment]
Thanks for the report. Your example works for me using Stata 14 and both save and saveold(13). Tomorrow I will seek for a PC with enough memory and Stata 13. The error you see is thrown because our function is trying to read a a string inside the dta-file, this string is either not (yet) written or we are for whatever kind of reason at a wrong position inside of the dta-file. Saving that much data from Stata took a while, maybe your dta-file was incompletely written or Stata 13 contains a bug we have to circumvent.

```R
> library("readstata13")
> dd <- read.dta13(file = "~/test13.dta")
> dim(dd)
[1] 1000000000          1
> attr(dd, "version")
[1] 117
```

**Comment by sjewo** [comment]
I can confirm this error on Windows 7 64bit with R 3.2.2 and readstata 0.8 with a Stata 14 test file. We take a look into this issue.

Could you please tell us about OS and R Version?

**Comment by pbaylis** [comment]
Thanks for looking into it, I'm on Windows 2008 Server with R 3.2.2 and
readstata 0.8 with a Stata 13 test file.

**Comment by pbaylis** [comment]
I also checked on OS X 10.10.5 (Yosemite) with Stata 13 and R 3.1.3/3.2.2 (I upgraded). The error did not occur, so perhaps this a Windows issue?

Later today I can try transporting the file saved in Stata 13 (Windows) to OS X and vice-versa to see if the problem is in the reading or the writing.

**Comment by sjewo** [comment]
We did some testing (awful slow because of small memory systems) and added some changes to fix this issue. Could you test it on your system with branch "bigfile"?
```
devtools::install_github("sjewo/readstata13", ref="bigfile")
library(readstata13)
df <- read.dta13("test.dta")
```

**Comment by pbaylis** [comment]
Still getting the same error:

     > df <- read.dta13("test.dta")
     Error: When attempting to read <strls>: Something went wrong!
     > packageVersion("readstata13")
     [1] ‘0.8.1’

**Comment by sjewo** [comment]
Hi, maybe you fetched the wrong branch (testing)? With branch "bigfile" you should see some additional warning with debug output if this error occurs.

**Comment by pbaylis** [comment]
I fetched the correct branch, but I do have another R instance open running something time-consuming. I'm not familiar with how the update procedure works, maybe it wasn't able to update the package due to the other session? I tried on a different windows computer (with Windows 7) and it's now working for me. 

I'll close but will try again on 2008 Server when my other task completes and report back if it doesn't work. Thanks!
